### ISSUE TITLE
Broken kubectl command and missing information

### ISSUE THREAD

**Issue by eb-trigo** [actions-runner-controller/actions-runner-controller#201]
Hey, guys!

I'm facing a problem with the latest version of your amazing **actions-runner-controller** tool. I hope you can help me.
 
When I run this command (`kustomize build . | kubectl apply -f -`) I get the following error.
**Error from server (NotFound): error when creating "STDIN": namespaces "actions-runner-system" not found:**

It's important to note that the **actions-runner-system** pods, namespace including runners agents currently running in a separate Kubernetes cluster from applications that don't contain this **actions-runner-system** namespace.

Another question I've is whether I should add **namespace: actions-runner-system** to my **kind: RunnerDeployment** & **kind: HorizontalRunnerAutoscaler** because when I deployed your [RunnerDeployment](https://github.com/summerwind/actions-runner-controller#runnerdeployments) with a default namespace I received 
 an error from the **controller-manager** pod saying the runners were not found in the same 
**actions-runner-system** namespace.

**Update:**
I was able to solve the namespace issue by explicitly mentioning `namespace: default` in my deployment YAML.
I still don't know what's causing the issue though, and where this is coming from?

**Comment by mumoshu** [comment]
@eb-trigo Hey! Just to be extra sure, in which directory are you running `kustomize build . | kubectl apply -f -`?

**Comment by eb-trigo** [comment]
@mumoshu I run it within one of the microservices app directories that contain Deployment yaml file.

It is important to note that in the previous versions of **actions-runner-controller** I didn't encounter this error only after the upgrade to the latest version.

**Comment by mumoshu** [comment]
This seems to be due to a lack of user's understanding how K8s manifests are written and installed.

`namespaces "actions-runner-system" not found` means your manifest contains `namespace: actions-runner-system` or you've specified `-n actions-runner-system` on `kubectl-apply`. You need to either run `kubectl create namespace actions-runner-controller` or create the namespace from a `kind: Namespace` resource manifest.

If it did work before, you had the namespace in that cluster, but not now, or you actually used other namespace to install before and using another namespace now.

**Comment by mumoshu** [comment]
Also note that our example kustomize config does produce the required namespace manifest for you.

```
$ kustomize build config/default | head
apiVersion: v1
kind: Namespace
metadata:
  labels:
    control-plane: controller-manager
  name: actions-runner-system
#snip
```
